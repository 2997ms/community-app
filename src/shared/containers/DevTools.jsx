/**
 * The DevTools container provides Redux monitors for use in development
 * environment.
 */

/* As this component is intended for development environment only, it is fine
 * to use dev-only dependencies. */
/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { createDevTools } from 'redux-devtools';
import DockMonitor from 'redux-devtools-dock-monitor';
import LogMonitor from 'redux-devtools-log-monitor';

export default createDevTools((
  /* Additional Redux monitors can be added here. */
  <DockMonitor
    changePositionKey="ctrl-p"
    toggleVisibilityKey="ctrl-m"
  >
    <LogMonitor theme="tomorrow" />
  </DockMonitor>
));
