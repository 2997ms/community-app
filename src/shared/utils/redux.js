/**
 * Redux-related helpers.
 */

import _ from 'lodash';

/**
 * Given any Flux Standard Action (FSA) with promise as the payload, it returns
 * a promise which resolves into the FSA result object.
 * @param {Object} action
 * @return Promise which resolves to the operation result.
 */
export function toFSA(action) {
  return action.payload.then(data => ({
    ...action,
    payload: data,
  }), error => ({
    ...action,
    payload: error,
    error: true,
  }));
}

/**
 * Given a mapping between reducer names and their promises, this function
 * waits till resolution of all promises and returns the mapping between
 * reducer names and the resolved reducers.
 * @param {Object} promises Object with promises of reducers.
 * @return Object with reducers.
 */
export function resolveReducers(promises) {
  return Promise.all(_.values(promises)).then((reducers) => {
    const map = {};
    _.keys(promises).forEach((key, index) => {
      map[key] = reducers[index];
    });
    return map;
  });
}

/* We don't any `main` function in this module to export it by default. */
export default undefined;
