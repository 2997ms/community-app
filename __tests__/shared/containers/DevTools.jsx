import DevTools from 'containers/DevTools';
import React from 'react';
import Rnd from 'react-test-renderer/shallow';
import { createStore } from 'redux';

const rnd = new Rnd();
const store = createStore(() => ({}), {}, DevTools.instrument());

test('Snapshot match', () => {
  rnd.render((
    <DevTools
      store={store}
    />
  ));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
