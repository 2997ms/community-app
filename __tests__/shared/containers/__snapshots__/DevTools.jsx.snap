// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<Connect(DockMonitor)
  changePositionKey="ctrl-p"
  defaultIsVisible={true}
  defaultPosition="right"
  defaultSize={0.3}
  fluid={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
  toggleVisibilityKey="ctrl-m"
>
  <LogMonitor
    expandActionRoot={true}
    expandStateRoot={true}
    markStateDiff={false}
    preserveScrollTop={true}
    select={[Function]}
    theme="tomorrow"
  />
</Connect(DockMonitor)>
`;
